bluetooth/src/commonMain/kotlin/Attribute.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ Copyright (c) 2020. Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.bluetooth
+
+import com.splendo.kaluga.base.flow.HotFlowable
+import com.splendo.kaluga.bluetooth.device.DeviceAction
+import com.splendo.kaluga.bluetooth.device.DeviceState
+import com.splendo.kaluga.state.StateRepo
+
+abstract class Attribute<R : DeviceAction.Read, W : DeviceAction.Write>(initialValue: ByteArray? = null, protected val stateRepo: StateRepo<DeviceState>) : HotFlowable<ByteArray?>(initialValue) {
+    abstract val uuid: UUID
+
+    suspend fun readValue() {
+        addAction(createReadAction())
```

Comment:
Not all characteristics allow value to read, how it will handled?

Reply:
Permissions are definitely lacking. Its been a while since I wrote this, but I looked at them and didnt implement them probably because I couldnt figure out how to bridge iOS and Android permisssions here. Maybe yu and @thoutbeckers can have a look as well.

Reply:
Its not just permissions btw, we should also account for encryption and authorization I think